src/parser_v3.py

```diff
@@ -0,0 +1,616 @@
+STATE_TEST_DATA = INPUT_DIR / 'state_test_data.json'
+## For adding metadata
+# For state notes and last updated
+STATE_WISE = INPUT_DIR / 'data.json'
+# For district notes
+DISTRICT_WISE = INPUT_DIR / 'state_district_wise.json'
+
+OUTPUT_DIR = Path('tmp', 'v3')
+OUTPUT_MIN_DIR = OUTPUT_DIR / 'min'
+OUTPUT_DATA_FILENAME = 'data'
+OUTPUT_TIMESERIES_FILENAME = 'timeseries'
+
+STATE_CODES = {
+    'Andhra Pradesh': 'AP',
+    'Arunachal Pradesh': 'AR',
+    'Assam': 'AS',
+    'Bihar': 'BR',
+    'Chhattisgarh': 'CT',
+    'Goa': 'GA',
+    'Gujarat': 'GJ',
+    'Haryana': 'HR',
+    'Himachal Pradesh': 'HP',
+    'Jharkhand': 'JH',
+    'Karnataka': 'KA',
+    'Kerala': 'KL',
+    'Madhya Pradesh': 'MP',
+    'Maharashtra': 'MH',
+    'Manipur': 'MN',
+    'Meghalaya': 'ML',
+    'Mizoram': 'MZ',
+    'Nagaland': 'NL',
+    'Odisha': 'OR',
+    'Punjab': 'PB',
+    'Rajasthan': 'RJ',
+    'Sikkim': 'SK',
+    'State Unassigned': 'UN',
+    'Tamil Nadu': 'TN',
+    'Telangana': 'TG',
+    'Tripura': 'TR',
+    'Uttarakhand': 'UT',
+    'Uttar Pradesh': 'UP',
+    'West Bengal': 'WB',
+    'Andaman and Nicobar Islands': 'AN',
+    'Chandigarh': 'CH',
+    'Dadra and Nagar Haveli and Daman and Diu': 'DN',
+    'Delhi': 'DL',
+    'Jammu and Kashmir': 'JK',
+    'Ladakh': 'LA',
+    'Lakshadweep': 'LD',
+    'Puducherry': 'PY',
+    'Total': 'TT',
+    # To accomodate for improper entries in old data
+    #  'Dadra and Nagar Haveli': 'DN',
+    #  'Daman And Diu': 'DN',
+}
+
+DISTRICTS_DICT = defaultdict(dict)
+DISTRICTS_ADDITIONAL = {
+    'bsf camp': 'BSF Camp',
+    'italians': 'Italians',
+    'other state': 'Other State',
+    'railway quarantine': 'Railway Quarantine',
+    'airport quarantine': 'Airport Quarantine',
+    'evacuees': 'Evacuees',
+    'foreign evacuees': 'Foreign Evacuees',
+    'unassigned': 'Unassigned',
+}
+
+PRINT_WIDTH = 60
+
+ddict = lambda: defaultdict(ddict)
+data = ddict()
+timeseries = ddict()
+
+
+def inc(ref, key, count):
+    if not isinstance(ref[key], int):
+        ref[key] = 0
+    ref[key] += count
+
+
+def parse(raw_data, i):
+    for j, entry in enumerate(raw_data['raw_data']):
+        state_name = entry['detectedstate']
+        try:
+            state = STATE_CODES[state_name]
+        except KeyError:
+            # Entries having unrecognized state names are discarded
+            if state_name:
+                print('[{}: {}] [{}] [Bad state: {}] {}'.format(
+                    i, j + 2, entry['dateannounced'], state_name,
+                    entry['numcases']))
+            continue
+
+        try:
+            fdate = datetime.strptime(entry['dateannounced'], '%d/%m/%Y')
+            date = datetime.strftime(fdate, '%Y-%m-%d')
+        except ValueError:
+            # Bad date
+            print('[{}: {}] [Bad date: {}] {}: {} {}'.format(
+                i, j + 2, entry['dateannounced'], entry['detectedstate'],
+                entry['detecteddistrict'], entry['numcases']))
+            continue
```

Comment:
Can this be part of a log file with error levels? Use logger.

Reply:
That can be done, but understand that all the print statements here are just warnings of something unexpected from the sheet. Since the script shouldn't crash, there's no error log per se. At least, unless something terrible happens like the API format changing or one of the input files is missing. But I'll look into adding logger.